syntax = "proto3";
import "replicate.proto";
package huawei.proto;

message VolumeMeta{
    VolumeInfo info = 1;
    repeated OperationRecord records = 2;
}
// volume status
enum VolumeStatus{
    VOL_UNKNOWN = 0;
    VOL_ENABLING = 1;
    VOL_AVAILABLE = 2;
    VOL_ATTACHED = 3;
    VOL_DELETED = 4;
}
message VolumeInfo{
    string vol_id = 1;
    string path = 2;
    VolumeStatus vol_status = 3;
    uint64 size = 4;
    bool rep_enable = 5;
    string rep_uuid = 6;
    repeated string peer_volumes = 7;
    RepRole role = 8;
    RepStatus rep_status = 9;
}

